### DISCUSSION TITLE
rename org.eclipse.osgi to org.eclipse.equinox.framework

### DISCUSSION THREAD

**Discussion by laeubi** [eclipse-equinox/equinox#283] [Ideas]
Currently we have the bundle (and it uses the name as artifact id as well) `org.eclipse.osgi`, while this historically might be a good choice it has become confusing over time and actually violates the Eclipse rules for project naming (and collides with https://projects.eclipse.org/projects/technology.osgi):

https://www.eclipse.org/projects/handbook/#resources-maven

I therefore like to suggest to take actions to migrate to a new name `org.eclipse.equinox.framework`, as a first step one could adjust the maven deployment to deploy it to a new artifact id and install a relocation pom:

https://maven.apache.org/guides/mini/guide-relocation.html

Compare for example:

- https://mvnrepository.com/artifact/org.apache.felix/org.apache.felix.framework
- https://mvnrepository.com/artifact/org.knopflerfish/framework

Also try to search on central for:

- felix framework (first hit)
versus
- equinox framework (no results)

**Comment by merks** [comment]
The naming issue is a red herring because the OSGi project uses the org.osgi namespace:

https://github.com/osgi/osgi

Also, the Eclipse project happily used org.eclipse.anything whenever the mood strikes without regard whether "anything" is a project or not.

Of course lots of things depend on that bundle name:

![image](https://github.com/eclipse-equinox/equinox/assets/208716/f8641c7a-bd64-4fab-a66c-58f5af723f20)

So changing that seems a bit of a non-starter...

So that leaves the question of "confusion".  I don't argue it's not a confusing name, though it seems not so confusing, but moving it is more likely to cause new confusion than the eliminate old confusion.   

One current finds it here:

https://repo1.maven.org/maven2/org/eclipse/platform/org.eclipse.osgi/

Will changing that help someone or create work for someone and maybe create new confusion about the two places it can be found?

**Reply by laeubi** [reply]
Equinox has to use `org.eclipse.equinox` and OSGi project is allowed to use `org.eclipse.osgi` thats the rules regardles if they *can* use other group IDs, this will change see: 
- https://github.com/osgi/osgi/discussions/596

beside that, I always need to remember that the Equinox **Framework(!)** does not named framework but "org.eclipse.osgi"  that really confusing and makes it hard to use in maven.

**Comment by merks** [comment]
Apparently Equinox has used these:

![image](https://github.com/eclipse-equinox/equinox/assets/208716/2aa5fde7-9710-4b7d-9a61-ec978b5892f3)

**Reply by laeubi** [reply]
Yes as I said, for historic reasons and it actually violates the namespace rules, equinox has namespace `org.eclipse.equinox` so we better get rid of it (some of your mentioned things are already worked on or deprecated or both).

**Comment by merks** [comment]
I hope that the cure is not worse than disease that we have lived with for 20 years...

**Comment by tjwatson** [comment]
`org.eclipse.osgi` as a bundle name is considered API.  Changing it will break an unknown (I think large) amount of things.

**Reply by laeubi** [reply]
> `org.eclipse.osgi` as a bundle name is considered API. Changing it will break an unknown (I think large) amount of things.

Should not all packages be internal or exposing standard osgi api? So consumers better use import-package anyways.

**Reply by tjwatson** [reply]
Package names are different from bundle names.  You want to change the bundle name, that is an API breaking change.  Changing API package names (or not exporting Eclipse API packages) like `org.eclipse.osgi.service.localization` would also be a major breaking change.
